provider-integration/im2/pkg/im/controller/manager.go

```diff
@@ -184,6 +239,7 @@ func sendResponseOrError(w http.ResponseWriter, data any, err error) {
 }
 
 func sendError(w http.ResponseWriter, err error) {
+	metricRequestErrorCounter.Inc()
```

Comment:
I think we should differentiate a bit more between the errors. At least if it is 4xx or 5xx.